docs/dev/authentication-authorization.md

```diff
@@ -0,0 +1,52 @@
+# Authentication and Authorization in Eclipse PASS
+
+## Authentication
+
+### User Interface Authentication
+
+Authentication for the user interface occurs through the use of an authentication service provider (SP), [pass-auth](https://github.com/eclipse-pass/pass-auth), written as a node application employing the express framework and acting as a reverse proxy in front of pass-core and other api services.
+
+`pass-auth` is configured to initiate a SAML exchange with a known identity provider (IDP) that supports [Shibboleth](https://shibboleth.atlassian.net/wiki/spaces/CONCEPT/overview). Although `pass-auth` itself is not a Shibboleth service provider specifically, it is a generalized SAML service provider that can handle specific Shibboleth interactions with an IDP. In response to a valid `authn` assertion against an IDP, `pass-auth` expects to receive and validate a Shibboleth SAML assertion against its assertion consumer service (ACS) URL. This assertion is expected to contain the following Shibboleth attributes:
+
+```
+'urn:oid:2.16.840.1.113730.3.1.241': 'displayName'
+'urn:oid:1.3.6.1.4.1.5923.1.1.1.9': 'scopedAffiliation'
+'urn:oid:0.9.2342.19200300.100.1.3': 'email'
+'urn:oid:2.16.840.1.113730.3.1.3': 'employeeNumber'
+'urn:oid:1.3.6.1.4.1.5923.1.1.1.1': 'employeeIdType'
+'urn:oid:1.3.6.1.4.1.5923.1.1.1.6': 'eppn'
+'urn:oid:2.5.4.42': 'givenName'
+'urn:oid:2.5.4.4': 'surname'
+'urn:oid:1.3.6.1.4.1.5923.1.1.1.13': 'uniqueId'
+'urn:oid:0.9.2342.19200300.100.1.1': 'uniqueIdType'
+```
+
+These Shibboleth attributes are used to locate a user in `pass-core` and set up a user object on the session. Initially, `pass-auth` will use the `eppn` attribute to locate a user in `pass-core` via a request that uses basic authentication. On all subsequent requests to `pass-core`, `pass-auth` will add the following headers using the Shibboleth attributes stored on a session:
```

Comment:
I'm a little confused by this. Why does pass-auth need to locate a PASS user? Suppose this is the first time a user logs into pass. There will be no pass user for pass-auth to find. The user will only be created when a request comes in with the shib attributes.

Reply:
This is a great point re the non-existent user. I think we can move this to a shibb header style request.

Generally, I think we want to get/create a user in some fashion just to ensure someone can use the system, but also provide the user to `pass-ui` so that `pass-ui` can use it for setting up the current-user service.